corehq/apps/data_dictionary/tests/utils.py

```diff
@@ -1,10 +1,14 @@
-from corehq.apps.data_dictionary.models import CaseType, CaseProperty
+from corehq.apps.data_dictionary.models import CaseType, CaseProperty, CasePropertyAllowedValue
 
 
-def setup_data_dictionary(domain, case_type_name, prop_list=None):
+def setup_data_dictionary(domain, case_type_name, prop_list=None, allowed_values=None):
     prop_list = prop_list or []
+    allowed_values = allowed_values or {}
     CaseType.get_or_create(domain, case_type_name)
     for prop_name, data_type in prop_list:
         prop = CaseProperty.get_or_create(prop_name, case_type_name, domain)
         prop.data_type = data_type
         prop.save()
+        if prop_name in allowed_values:
+            for value in allowed_values[prop_name]:
+                CasePropertyAllowedValue.objects.create(case_property=prop, allowed_value=value)
```

Comment:
can this be done in bulk? Also should this be a `get_or_create`?

Reply:
yeah, `get_or_create` makes sense. I could go either way on the bulk save, as there aren't likely to be that many values. @kmtracey could you update this?

Reply:
Updated in https://github.com/dimagi/commcare-hq/pull/30786